Source/WebCore/Modules/gamepad/GamepadHapticActuator.cpp

```diff
@@ -97,11 +105,12 @@ void GamepadHapticActuator::playEffect(EffectType effectType, GamepadEffectParam
 
     effectParameters.duration = std::min(effectParameters.duration, GamepadEffectParameters::maximumDuration.milliseconds());
 
-    m_playingEffectPromise = WTFMove(promise);
-    GamepadProvider::singleton().playEffect(m_gamepad->index(), m_gamepad->id(), effectType, effectParameters, [this, protectedThis = makePendingActivity(*this), playingEffectPromise = m_playingEffectPromise](bool success) mutable {
-        if (m_playingEffectPromise != playingEffectPromise)
+    currentEffectPromise = WTFMove(promise);
```

Comment:
Since we move the `promise` argument to the local `currentEffectPromise` variable, but don't move that into the playEffect closure, could this go out of scope and cause problems? Previously it was moved to the `m_playingEffectPromise`, which seems like it would keep it alive.

Reply:
`currentEffectPromise` is declared like so:
```
auto& currentEffectPromise = promiseForEffectType(effectType);
```

and it returns a reference to the right promise data member. I didn't change any lifetime here. The promise still gets stored as RefPtr in a data member. It is merely abstracted away by the `currentEffectPromise` C++ reference since we have more than one such data member now.